### ISSUE TITLE
"use Enter to stop" sbt feature breaks in Play 2.8.11

### ISSUE THREAD

**Issue by jaykaysharewell** [playframework/playframework#11032]
### Play Version

2.8.11

### API

Java

### Operating System

macOS Catalina 12.0.1

### JDK

openjdk 11.0.11 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)

### Expected Behavior

command line: `JAVA_HOME="/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home" sbt run`

output:

```
--- (Running the application, auto-reloading is enabled) ---
[info] p.c.s.AkkaHttpServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000
(Server started, use Enter to stop and go back to the console...)
```
**Expected behavior: this works correctly in Play 2.8.8**, pressing Enter should quit the server and produce this output:

```
[info] p.a.i.l.c.CoordinatedShutdownSupport - Starting synchronous coordinated shutdown with ServerStoppedReason reason and 35000 milliseconds timeout
[info] p.c.s.AkkaHttpServer - Stopping Akka HTTP server...
[info] p.c.s.AkkaHttpServer - Terminating server binding for /0:0:0:0:0:0:0:0:9000
[info] p.c.s.AkkaHttpServer - Running provided shutdown stop hooks
```

### Actual Behavior

Actual behavior: **this is broken in Play 2.8.11**, pressing Enter has no effect. Only Ctrl-C can quit the server

**Comment by mkurz** [comment]
Confirmed, this is broken now :disappointed: 
@gmethvin Very likely this is caused by #10890, can you have a look?

**Comment by gmethvin** [comment]
I'm not sure I'll get a chance in the next week but I'll try to take a look as soon as I can. It's not obvious to me what the specific issue is.

**Comment by mkurz** [comment]
No problem, if I find time over the weekend I'll try to track it down.

**Comment by mkurz** [comment]
From @JustinPihony 
> They were able to trace it to a change from sbt 1.3.9 to 1.4.0+ when using the newest Play

**Comment by aldenml** [comment]
I can confirm that if https://github.com/playframework/playframework/pull/10890 is reverted, regular `sbt run` with sbt >= 1.4.0 works as expected. I tried with a modified local build of `sbt-plugin`.

**Comment by uncle-betty** [comment]
When `createReader` is invoked, SBT has already changed `System.in` to be the `proxyInputStream` object from `sbt.internal.util.Terminal`. At least that's what it looks like to me, based on a little bit of poking around with `jdb`.

`proxyInputStream` extends `SimpleInputStream` (to be found in the same source code file), which in turn extends `InputStream`. Neither `proxyInputStream` nor `SimpleInputStream` seems to override `InputStream`'s `available()` method.

So, as far as I can see, `proxyInputStream` - and thus `System.in` - uses `InputStream`'s implementation of `available()`, which always returns 0.

I suspect that this might be what breaks the polling mechanism. I have no idea how to fix this, but I figured that collecting a little more data on this issue couldn't hurt.

**Comment by mkurz** [comment]
I found out what the problem is, PR will follow.

**Comment by mkurz** [comment]
Fix is here: #11061

**Comment by mkurz** [comment]
Fixed.
